core/geometry/src/clipping/ClipUtils.ts

```diff
@@ -625,34 +627,111 @@ export class ClipUtilities {
     }
     return false;
   }
+  /**
+   * Create a clipper from the transformed range.
+   * @param range input range to create clipper from
+   * @param transform how to transform the range (NOTE: applied to the range faces without swelling the range volume)
+   * @param degeneratePoints optionally populated with the 1 or 2 points defining the transformed range if it is degenerate (all points colinear/coincident); otherwise untouched
+   * @returns newly constructed clipper. If no clip planes could be computed, fill `degeneratePoints` and return undefined.
+  */
+  private static createClipperFromTransformedRange3d(range: Range3d, transform: Transform, degeneratePoints?: Point3d[]): ConvexClipPlaneSet | undefined {
+    if (!transform)
+      transform = Transform.createIdentity();
+    const builder = PolyfaceBuilder.create();
+    builder.addTransformedRangeMesh(transform, range);
+    const mesh = builder.claimPolyface();
+    const clipper = this._workClipper = ConvexClipPlaneSet.createConvexPolyface(mesh, this._workClipper).clipper;
+    if (clipper.planes.length > 0)
+      return clipper;
+    // no faces found in the compressed mesh
+    if (degeneratePoints) {
+      assert(mesh.data.point.length <= 2);
+      for (let i = 0; i < 2; ++i) {
+        const point = mesh.data.point.getPoint3dAtCheckedPointIndex(i);
+        if (point)
+          degeneratePoints.push(point);
+      }
+    }
+    return undefined;
+  }
+  /**
+   * Handle pathological cases of range-range intersection, where one of the ranges defines no area or volume (is a line segment or single point).
+   * @param range local range to intersect with the point/segment
+   * @param points isolated local point, or local segment
```

Comment:
so you expect the size of points array is 1 or 2. If it's more than 2 you just ignore extra elements? If so, I suggest this:
```suggestion
   * @param points isolated local point, or local segment's start and end
```